src/ucp/core/ucp_request.h

```diff
@@ -152,11 +152,14 @@ struct ucp_request {
 
             };
 
+            struct {
```

Comment:
add comment that this struct holds all mutable fields, and everything else except maybe flags is immutable